### ISSUE TITLE
Add endColumn to no-useless-escape

### ISSUE THREAD

**Issue by YodaDaCoda** [eslint/eslint#11629]
**What rule do you want to change?** no-useless-escape

**Does this change cause the rule to produce more or fewer warnings?** no

**How will the change be implemented? (New option, new default behavior, etc.)?** New default behaviour: return an endColumn

**Please provide some example code that this change will affect:**

```js
var regex = /[\.]/;
```

**What does the rule currently do for this code?** Currently, the rule correctly detects an issue with the code. However, since the rule doesn't return an endColumn, my IDE ([Atom](https://atom.io/)) highlights until the end of the line in this example (from my testing, I think it highlights until the end of the line, or until the next `\`), which makes for a rather confusing UX.

**What will the rule do after it's changed?** Return an endColumn

**Are you willing to submit a pull request to implement this change?** No

**Comment by platinumazure** [comment]
Thanks @YodaDaCoda for creating a new issue!

Personally I'm on the fence about whether this should be a bug fix or an enhancement. In any event, I will champion this change.

**Comment by kaicataldo** [comment]
This is now accepted.
